frontend/src/components/activePage/Toolbox/Toolbox.tsx

```diff
@@ -0,0 +1,257 @@
+  showCheckbox: boolean;
+  setShowCheckbox: Dispatch<SetStateAction<boolean>>;
+  setShowAlert: Dispatch<SetStateAction<boolean>>;
+  selectiveDestroy: Array<string>;
+  deselectAll: () => void;
+}
+
+const Toolbox: FC<IToolboxProps> = ({ ...props }) => {
+  const {
+    setSearchField,
+    setExpandAll,
+    setCollapseAll,
+    showAdvanced,
+    setShowAdvanced,
+    showCheckbox,
+    setShowCheckbox,
+    setShowAlert,
+    selectiveDestroy,
+    deselectAll,
+  } = props;
+  const [toolboxPopover, setToolboxPopover] = useState(false);
+  const [searchPopover, setSearchPopover] = useState(false);
+
+  const mobileContent = (
+    <div className="flex flex-col justify-center gap-2">
+      <div className="flex justify-between gap-2 md:hidden">
+        <Button
+          type="primary"
+          shape="round"
+          size="middle"
+          icon={<FullscreenOutlined />}
+          onClick={() => setExpandAll(true)}
+        >
+          Expand
+        </Button>
+        <Button
+          type="primary"
+          shape="round"
+          size="middle"
+          icon={<FullscreenExitOutlined />}
+          onClick={() => setCollapseAll(true)}
+        >
+          Collapse
+        </Button>
+      </div>
+      <Divider type="horizontal" className="my-2 md:hidden" />
+      <div className="flex flex-col justify-start gap-2 xl:hidden">
+        <div className="flex items-center gap-2">
+          <Switch
+            checked={showAdvanced}
+            onClick={e => setShowAdvanced(e)}
+            size="small"
+          />
+          <span>Show Header</span>
+        </div>
+        <div className="flex items-center gap-2">
+          <Switch
+            checked={showCheckbox}
+            onChange={e => setShowCheckbox(e)}
+            size="small"
+          />
+          <span>Show Checkbox</span>
+        </div>
+      </div>
+
+      <Divider type="horizontal" className="my-2 xl:hidden" />
+      <Button
+        type="primary"
+        ghost
+        shape="round"
+        size="middle"
+        icon={<DeleteOutlined />}
+        onClick={e => {
+          e.stopPropagation();
+          setToolboxPopover(false);
+          setShowAlert(true);
+        }}
+        disabled={!selectiveDestroy.length}
+      >
+        Destroy Selected{` (${selectiveDestroy.length})`}
+      </Button>
+      <Button
+        type={!selectiveDestroy.length ? 'primary' : 'warning'}
+        ghost
+        shape="round"
+        size="middle"
+        onClick={() => deselectAll()}
+        disabled={!selectiveDestroy.length}
+      >
+        Deselect All
+      </Button>
+    </div>
+  );
+  return (
+    <div className="h-full flex justify-start items-center gap-2 px-5 py-3">
+      <Search
+        allowClear
+        className="hidden sm:block"
+        placeholder="Search User"
+        onChange={event => setSearchField(event.target.value)}
+        onSearch={value => setSearchField(value)}
```

Comment:
```suggestion
        onSearch={setSearchField}
```
Check other occurrences :)